src/appendix/glossary.md

```diff
@@ -1,96 +1,491 @@
+an extension to Rust's borrowing system to make it be based on the control-flow
+graph.
+
+#### node-id or NodeId
+
+An index identifying a particular node in the AST or HIR; gradually being phased
+out and replaced with `HirId`. See [the HIR chapter for
+more](../hir.md#identifiers-in-the-hir).
+
+#### obligation
+
+Something that must be proven by the trait system. ([see
+more](../traits/resolution.md)) placeholder">placeholder **NOTE: skolemization
+is deprecated by placeholder** a way of handling subtyping around "for-all"
+types (e.g., `for<'a> fn(&'a u32)`) as well as solving higher-ranked trait
+bounds (e.g., `for<'a> T: Trait<'a>`). See [the chapter on placeholder and
+universes](../borrow_check/region_inference/placeholders_and_universes.md) for
+more details.
+
+#### point
+
+Used in the NLL analysis to refer to some particular location in the MIR;
+typically used to refer to a node in the control-flow graph.
+
+#### polymorphize
+
+An optimization that avoids unnecessary monomorphisation. ([see
+more](../backend/monomorph.md#polymorphization)) projection">projection A
+general term for a "relative path", e.g. `x.f` is a "field projection", and
+`T::Item` is an ["associated type
+projection"](../traits/goals-and-clauses.md#trait-ref).
+
+#### promoted constants
+
+Constants extracted from a function and lifted to static scope; see [this
+section](../mir/index.md#promoted) for more details.
+
+#### provider
+
+The function that executes a query. ([see more](../query.md))
+
+#### quantified
+
+In math or logic, existential and universal quantification are used to ask
+questions like "is there any type T for which is true?" or "is this true for all
+types T?"; see [the background chapter for more](./background.md#quantified).
+
+#### query
+
+A sub-computation during compilation. Query results can be cached in the current
+session or to disk for incremental compilation. ([see more](../query.md))
+
+#### recovery
+
+Recovery refers to handling invalid syntax during parsing (e.g. a missing comma)
+and continuing to parse the AST. This avoid showing spurious errors to the user
+(e.g. showing 'missing field' errors when the struct definition contains
+errors). region">region Another term for "lifetime" often used in the literature
+and in the borrow checker.
+
+#### rib
+
+A data structure in the name resolver that keeps track of a single scope for
+names. ([see more](../name-resolution.md))
+
+#### scrutinee
+
+A scrutinee is the expression that is matched on in `match` expressions and
+similar pattern matching constructs. For example, in `match x { A => 1, B => 2 }`, 
+the expression `x` is the scrutinee.
+
+#### sess
+
+The compiler session, which stores global data used throughout compilation
+side-tables">side tables Because the AST and HIR are immutable once created, we
+often carry extra information about them in the form of hashtables, indexed by
+the id of a particular node.
```

Comment:
Same here
```suggestion
#### side tables

Because the AST and HIR are immutable once created, we
often carry extra information about them in the form of hashtables, indexed by
the id of a particular node.
```